os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -111,9 +113,33 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 
 		mm_takesemaphore(heap);
 
-		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; node = (struct mm_allocnode_s *)((char *)node + node->size)) {
+		prev = NULL;
+		node = heap->mm_heapstart[region];
+		next = (struct mm_allocnode_s *)((char *)node + node->size);
+
+		for (; node < heap->mm_heapend[region]; prev = node, node = next, next = (struct mm_allocnode_s *)((char *)next + next->size)) {
 			mvdbg("region=%d node=%p size=%p preceding=%p (%c)\n", region, node, node->size, (node->preceding & ~MM_ALLOC_BIT), (node->preceding & MM_ALLOC_BIT) ? 'A' : 'F');
 
+			if ((prev && prev->size != node->preceding & ~MM_ALLOC_BIT) ||
+				(node->size != next->preceding & ~MM_ALLOC_BIT) ||
+				(!(node->preceding & MM_ALLOC_BIT) && !(prev->preceding & MM_ALLOC_BIT) && ((struct mm_freenode_s *)node)->blink != prev) || 
```

Comment:
This is either wrong. Free node is sorted by size, it's not by location. So, even both are free node, node -> blink could not be prev.

Reply:
And If both are free nodes, then they will be merged as one free node. This is checking for impossible case.

Reply:
Oh.. yes.. then we have to remove it..

Reply:
(((struct mm_freenode_s*)node)->blink->flink != node || ((struct mm_freenode_s *)node)->flink->blink != node))

Reply:
You can use this. This is what you want

Reply:
Done. Thanks.